cilium-cli/connectivity/builder/pod_to_ingress_service.go

```diff
@@ -4,16 +4,76 @@
 package builder
 
 import (
+	_ "embed"
+	"strings"
+
 	"github.com/cilium/cilium/cilium-cli/connectivity/check"
 	"github.com/cilium/cilium/cilium-cli/connectivity/tests"
 	"github.com/cilium/cilium/cilium-cli/utils/features"
 )
 
+var (
+	//go:embed manifests/allow-ingress-identity.yaml
+	allowIngressIdentityPolicyYAML string
+
+	//go:embed manifests/deny-ingress-backend.yaml
+	denyIngressBackendPolicyYAML string
+
+	//go:embed manifests/deny-ingress-entity.yaml
+	denyIngressIdentityPolicyYAML string
+
+	//go:embed manifests/deny-ingress-source-egress-other-node.yaml
+	denyIngressSourceEgressOtherNodePolicyYML string
+)
+
 type podToIngressService struct{}
 
-func (t podToIngressService) build(ct *check.ConnectivityTest, _ map[string]string) {
+func (t podToIngressService) build(ct *check.ConnectivityTest, templates map[string]string) {
 	// Test Ingress controller
 	newTest("pod-to-ingress-service", ct).
 		WithFeatureRequirements(features.RequireEnabled(features.IngressController)).
 		WithScenarios(tests.PodToIngress())
+
+	newTest("pod-to-ingress-service-allow-ingress-identity", ct).
+		WithFeatureRequirements(features.RequireEnabled(features.IngressController)).
+		WithCiliumPolicy(denyAllIngressPolicyYAML).
+		WithCiliumPolicy(allowIngressIdentityPolicyYAML).
+		WithScenarios(tests.PodToIngress())
+
+	newTest("pod-to-ingress-service-deny-all", ct).
+		WithFeatureRequirements(features.RequireEnabled(features.IngressController)).
+		WithCiliumPolicy(denyAllIngressPolicyYAML).
+		WithScenarios(tests.PodToIngress()).
+		WithExpectations(func(_ *check.Action) (egress check.Result, ingress check.Result) {
+			return check.ResultDefaultDenyEgressDrop, check.ResultNone
+		})
+
+	newTest("pod-to-ingress-service-deny-backend-service", ct).
+		WithFeatureRequirements(features.RequireEnabled(features.IngressController)).
+		WithCiliumPolicy(denyIngressBackendPolicyYAML).
+		WithScenarios(tests.PodToIngress()).
+		WithExpectations(func(_ *check.Action) (egress check.Result, ingress check.Result) {
+			return check.ResultDefaultDenyEgressDrop, check.ResultNone
+		})
+
+	newTest("pod-to-ingress-service-deny-ingress-identity", ct).
+		WithFeatureRequirements(features.RequireEnabled(features.IngressController)).
+		WithCiliumPolicy(denyIngressIdentityPolicyYAML).
+		WithScenarios(tests.PodToIngress()).
+		WithExpectations(func(_ *check.Action) (egress check.Result, ingress check.Result) {
+			return check.ResultDefaultDenyEgressDrop, check.ResultNone
+		})
+
+	newTest("pod-to-ingress-service-deny-source-egress-other-node", ct).
+		WithCiliumVersion(">1.17.1 >1.16.7 >1.15.14").
```

Comment:
@sayboras hmm something's odd here - I just saw a run with

> Skipping test [pod-to-ingress-service-deny-source-egress-other-node] [20/22] (requires Cilium version >1.17.1 >1.16.7 >1.15.14 but running 1.16.8)

Reply:
🤔 interesting, I validated in main version to make sure the test was run.

Let me take a look, it could be existing bug with version comparison.

Reply:
[This CLI change](https://github.com/cilium/cilium/pull/35742/files#diff-7a17abdbd1cb8e727bf901ec2001452944a48e4d03d83a46ee7e8e760fe9e6b3R34) I merged few months ago might help. Would write:

```go
WithCiliumVersion(">1.15.14 <1.16.0 || >1.16.7 < 1.17.0 || >1.17.1").
```

Reply:
oh thanks a lot, seems like the current expression `>1.17.1 >1.16.7 >1.15.14` is only for 1.17.x

Reply:
thanks again both

https://github.com/cilium/cilium/pull/38803